### ISSUE TITLE
Dataset(allstate ClaimPrediction Challenge) are different from papers-" lightgbm-a-highly-efficient-gradient-boosting-decision-tree"

### ISSUE THREAD

**Issue by Andychen2018** [Microsoft/LightGBM#2072]
The features of Allstate in paper"lightgbm-a-highly-efficient-gradient-boosting-decision-tree" are 4228,but i found 34 features  in https://www.kaggle.com/c/ClaimPredictionChallenge. 
And the objective of the experiment in this paper is  Binary classiﬁcation,but the objective in "ClaimPredictionChallenge" is different

**Comment by guolinke** [comment]
we use the same config for allstate data as xgboost paper, which uses the one-hot encoding for categorical features and expand the number of features to 4228. As for the objective, we use the same as xgboost as well.

**Comment by Andychen2018** [comment]
I want to reproduce this experiment.Can you give the code of data processing or smoe advice,thanks a lot

**Comment by guolinke** [comment]
@Andychen2018 sorry, I cannot find the code.

I follow this to preprocessing:

![image](https://user-images.githubusercontent.com/16040950/55274544-780bc780-5314-11e9-8017-1b5fefbaa9a2.png)
